### DISCUSSION TITLE
SoC reading

### DISCUSSION THREAD

**Discussion by erepeo** [lbbrhzn/ocpp#1447] [Ideas]
Is it possible to add a car SoC reading?
![Zrzut ekranu 2024-12-26 095805](https://github.com/user-attachments/assets/3d2e3aae-be00-4343-aef3-5f81aebcd24a)

**Comment by lbbrhzn** [comment]
It will only work if your car can report SoC to your charger, for instance through the ISO 15118 protocol.

**Reply by erepeo** [reply]
And how can I verify this? When charging on public chargers, the SoC is reported to the charger (CCS - maybe it's different than Type2) so that means I should see the SoC in Home Assistant?

**Comment by lbbrhzn** [comment]
You need to check the specifications of you charger and your car.  ISO 15118 is known as 'Plug and Charge'.
Note that DC charging is very different from AC charging.

**Reply by erepeo** [reply]
Hmm
http://immari.pl/materials/Elvi.pdf
I don't think I have that option. Can you recommend an EVSE that gives you that option and doesn't cost a fortune? Maybe a DIY solution?

**Comment by lbbrhzn** [comment]
The EVBox Liviqo 2 does seem to have at least hardware support for it:

![image](https://github.com/user-attachments/assets/e026b71e-078f-43aa-93e1-b9ad0a282097)

As a workaround, you could check if there is an integration for your car that reads the SoC from the cloud, or use an OBD2 reader with BLE.

**Reply by erepeo** [reply]
Interesting. Thank you.
As for car integration, it exists but it works so unreliably that I removed it. I prefer something that physically reads the SoC. I am interested in this topic.
